src/WorkspaceContext.ts

```diff
@@ -95,6 +85,45 @@ export class WorkspaceContext implements vscode.Disposable {
         return { dispose: () => this.observers.delete(fn) };
     }
 
+    // report swift version and throw error if it failed to find swift
+    async reportSwiftVersion() {
+        try {
+            const { stdout } = await execSwift(['--version']);
+            const version = stdout.trimEnd();
+            this.outputChannel.log(version);
+        } catch(error) {
+            throw Error("Cannot find swift executable.");
+        }
+    }
+
+    // find LLDB version and setup path in CoreLLDB
+    async setLLDBVersion() {
+        // don't set LLDB on windows as swift version is not working at the moment
+        if (process.platform === 'win32') { return; }
+        const libPath = await getLLDBLibPath(getSwiftExecutable('lldb'));
+        if (!libPath) { return; }
+
+        const lldbConfig = vscode.workspace.getConfiguration('lldb');
```

Comment:
We could also use our `configuration` object here.
Although the property model is a bit limited, so we'd need to add a `setLLDBLibrary` method instead of a property setter, so that you can also pass in the configuration target.
Which do you prefer?